### PR TITLE
Use YAML as default config file format

### PR DESCRIPTION
**Pull request by AvdN**
Further abstraction from ConfigParser, introduction of YAML as
(default) config file format.

- introduces NoOptionError, and have IniConfig.get() throw that and
  remove further cmdline.py and util.py dependency on ConfigParser
- steve.yaml/YamlConfig is now the default name/type for config file
- old projects (steve.ini) are still found and will be read
- parametrized test for config file creation/retrieval (on util.py level)
  for both config types

### CODE REVIEW

**Comment by AvdN** [comment]
I read the comment about ditching the ConfigParser support, but for **me** having this currently is helpful as I can more easily compare changes in behaviour (i.e. not having another virtualenv with some old version of steve-cmd installed, that has none or half of my changes)
In this redo of PR 47, I even further abstracted and the extra stuff for ConfigParser support is less than 100 lines: The class with its load/set/get method and the file template and extra few lines in util:with_config() to try and load the other format(s) than the default one. 
It is fine with me to drop out the IniConfig related stuff at some point, but the abstraction Config/YamlConfig is IMO a good think to keep, to prevent relying on internals too much.

**Comment by AvdN** [comment]
I should probably have put this on a branch in order not to update the PR and then explicitly make it a a new PR. Not sure what is the best way to proceed right now.
If you run ``tox`` on these changes, make sure to run ``tox -r`` at least once to get the changes in ``setup.py``.

The commit message from the second change set has some important info, partly replicated in the added sync.rst doc file.

Next would be to make fetch, webedit, and all the other commands except for push, directly work on the YAML files instead of on the JSON files.

Since yamldata is a thin layer on top of ruamel.yaml.convert.syncjson: SyncJSON, there are no additional tests (beyond the [ones for that package](https://bitbucket.org/ruamel/yaml.convert/src/078586fa60393b794cacbb64876efd71416aa2a1/_test/test_syncjson.py?at=default)) that check if the sync works as expected

File: docs/commandline.rst [hunk 1]
```diff
@@ -87,46 +87,52 @@ Example use
...
+      url: 
+    
+      # The projectpath is where steve assumes subdirs if not explitly set
+      # projectpath: /data1/src/pyvideo.steve/tmp
+      # The jsonpath, if set, is where steve will look for the JSON files
+      # jsonpath: /data1/src/pyvideo.steve/tmp/json
+      # The yamlpath, if set, is where steve will look for the YAML files
+      # yamlpath: /data1/src/pyvideo.steve/tmp/yaml
+    
+      # name of status file in project directory (should not be steve.yaml)
```
Comment (willkg) [inline docs/commandline.rst]: I'd leave the whole `steve.yaml` thing out. There's no need to mention it here.

File: docs/commandline.rst [hunk 2]
```diff
@@ -87,46 +87,52 @@ Example use
...
+    
+      # Your username and api key.
+      # e.g. username: willkg
+      #      api_key: OU812
+      # username:
+      # api_key:
+      #
+      # Alternatively, you can pass this on the command line or put it in a
+      # separate API_KEY file which you can keep out of version control.
+      # cred_file:
```
Comment (willkg) [inline docs/commandline.rst]: This seems like it's missing something.

**Comment by willkg** [comment]
I skimmed the rest, but didn't comment on it because this pull request adds a whole bunch of stuff that I don't understand the use case for and there's been no discussion of as far as I know. Generally, when someone goes off and builds a bunch of stuff and dumps it in one big pr, they run the risk of building things people don't want. I think this PR and PR #47 are trying to do too much all in one change. For me as a reviewer and project maintainer, it's kind of exhausting to try to work through the changes in regards to code, architecture *and* figuring out whether all the new features were things I'm interested in and have big impact on using steve to get things done.

I thought what you were tackling was changing steve to use YAML for the configuration file. I'm interested in a PR for that. For the additional features you want, you should open up an issue for each one and we can discuss what it is you're thinking and whether it makes sense to add.

Given that, I'm going to close this out.

**Pull request closed without merging**